services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_service_specs/sidecar.py

```diff
@@ -53,6 +54,16 @@ def _get_environment_variables(
     if scheduler_data.paths_mapping.state_exclude is not None:
         state_exclude = scheduler_data.paths_mapping.state_exclude
 
+    callbacks_mapping: CallbacksMapping = deepcopy(scheduler_data.callbacks_mapping)
```

Comment:
Q: why do you deepcopy this? this is a bit weird no? you basically always change the scheduler data???